File: src/tools/scc/eval-repos/synthetic/go/module/nested.go
// Package module provides nested functionality.
package module

import (
	"sync"
)

// NestedItem represents an item in the nested service.
type NestedItem struct {
	ID        int
	Name      string
	CreatedAt int64
}

// NestedService provides item management.
type NestedService struct {
	items []NestedItem
	mu    sync.RWMutex
}

// NewNestedService creates a new service.
func NewNestedService() *NestedService {
	return &NestedService{
		items: make([]NestedItem, 0),
	}
}

// Add adds an item.
func (s *NestedService) Add(item NestedItem) {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.items = append(s.items, item)
}

// Find searches for an item by ID.
func (s *NestedService) Find(id int) *NestedItem {
	s.mu.RLock()
	defer s.mu.RUnlock()

	for i := range s.items {
		if s.items[i].ID == id {
			return &s.items[i]
		}
	}
	return nil
}

// GetAll returns all items.
func (s *NestedService) GetAll() []NestedItem {
	s.mu.RLock()
	defer s.mu.RUnlock()

	result := make([]NestedItem, len(s.items))
	copy(result, s.items)
	return result
}

// Clear removes all items.
func (s *NestedService) Clear() int {
	s.mu.Lock()
	defer s.mu.Unlock()

	count := len(s.items)
	s.items = s.items[:0]
	return count
}

// NestedProcessor processes items.
type NestedProcessor interface {
	Process(item NestedItem) (bool, error)
}

// AsyncProcessor implements NestedProcessor.
type AsyncProcessor struct{}

// Process processes an item asynchronously.
func (p *AsyncProcessor) Process(item NestedItem) (bool, error) {
	return item.ID > 0, nil
}

File: src/tools/scc/eval-repos/synthetic/rust/module/nested.rs
//! Nested module demonstrating Rust module patterns.

use std::sync::{Arc, RwLock};

/// Represents a nested item.
#[derive(Debug, Clone)]
pub struct NestedItem {
    pub id: i32,
    pub name: String,
    pub created_at: i64,
}

/// Service for managing nested items.
#[derive(Debug, Default)]
pub struct NestedService {
    items: Arc<RwLock<Vec<NestedItem>>>,
}

impl NestedService {
    /// Creates a new service.
    pub fn new() -> Self {
        NestedService {
            items: Arc::new(RwLock::new(Vec::new())),
        }
    }

    /// Adds an item.
    pub fn add(&self, item: NestedItem) {
        let mut items = self.items.write().unwrap();
        items.push(item);
    }

    /// Finds an item by ID.
    pub fn find(&self, id: i32) -> Option<NestedItem> {
        let items = self.items.read().unwrap();
        items.iter().find(|i| i.id == id).cloned()
    }

    /// Gets all items.
    pub fn get_all(&self) -> Vec<NestedItem> {
        let items = self.items.read().unwrap();
        items.clone()
    }

    /// Clears all items.
    pub fn clear(&self) -> usize {
        let mut items = self.items.write().unwrap();
        let count = items.len();
        items.clear();
        count
    }
}

/// Trait for processing items.
pub trait NestedProcessor {
    fn process(&self, item: &NestedItem) -> bool;
}

/// Async processor implementation.
pub struct AsyncProcessor;

impl NestedProcessor for AsyncProcessor {
    fn process(&self, item: &NestedItem) -> bool {
        item.id > 0
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_nested_service() {
        let service = NestedService::new();
        service.add(NestedItem {
            id: 1,
            name: "Test".to_string(),
            created_at: 0,
        });
        assert!(service.find(1).is_some());
    }
}
